software/firmware/europi_config.py

```diff
@@ -6,20 +6,85 @@
 PICO_DEFAULT_CPU_FREQ = 125_000_000
 OVERCLOCKED_CPU_FREQ = 250_000_000
 
+# Moog/Eurorack standard is 1.0 volts per octave
+MOOG_VOLTS_PER_OCTAVE = 1.0
+
+# Buchla standard is 1.2 volts per octave (0.1 volts per semitone)
+BUCHLA_VOLTS_PER_OCTAVE = 1.2
+
 
 class EuroPiConfig:
-    """This class provides EuroPi's global config points."""
+    """This class provides EuroPi's global config points.
+
+    To override the default values, create /config/config_EuroPiConfig.json on the Raspberry Pi Pico
+    and populate it with a JSON object. e.g. if your build has the oled mounted upside-down compared
+    to normal, the contents of /config/config_EuroPiConfig.json should look like this:
+
+    {
+        "rotate_display": true
+    }
+    """
 
     @classmethod
     def config_points(cls):
+        # fmt: off
         return [
-            configuration.choice(name="pico_model", choices=["pico", "pico w"], default="pico"),
+            # EuroPi revision -- this is currently unused, but reserved for future expansion
+            configuration.choice(
+                name="europi_model",
+                choices = ["europi", "europi_lc", "europi_x"],
+                default="europi"
+            ),
+
+            # CPU & board settings
+            configuration.choice(
+                name="pico_model",
+                choices=["pico", "pico w"],
+                default="pico"
+            ),
             configuration.choice(
                 name="cpu_freq",
                 choices=[PICO_DEFAULT_CPU_FREQ, OVERCLOCKED_CPU_FREQ],
                 default=OVERCLOCKED_CPU_FREQ,
             ),
+
+            # Display settings
+            configuration.choice(
+                name="rotate_display",
+                choices=[False, True],
+                default=False
+            ),
+            configuration.integer(
+                name="display_width",
+                range=range(8, 1024),
+                default=128
+            ),
+            configuration.integer(
+                name="display_height",
+                range=range(8, 1024),
+                default=32
+            ),
+
+            # Synthesizer family settings
+            # Normally this is intended for Eurorack compatibility, but being open-source someone may
+            # want to use it in an ecosystem that uses different specs
+            configuration.choice(
+                name="volts_per_octave",
```

Comment:
This might be something that requires a larger discussion about the config menu, because I can see that some users might set this value and then be confused when a script that doesn't take it into account outputs the wrong pitch voltage level.

For that reason, even though it's really annoying to have to repeat code in each script, I think the config menu should probably be kept to only firmware parameters, so that a user setting that value knows *for sure* that it'll set it across the board, for every script.

If the code you've got for switching between 1V/Oct and 1.2V/Oct is capable of it, maybe this would be something that could go in its own PR to add to the firmware? That way the parameter can still be included in config, and every script can innately make use of it. I'm imagining that you'd just want to call ain.pitch() and receive a float in semitones away from zero.

Reply:
I agree that this setting should apply across the module and that it would be inconvenient for both users and developers to have to set it in each script that uses quantization. 

The only problem i have is referencing experimental code or features from within the firmware proper. The experimental directory should provide its own set of configurations separate from that provided by the europi firmware. The downside of this, of course, is that users will have to reconfigure if and when a config point moves from experimental to firmware. We could probably solve that issue with some migration code if it was a problem.

@roryjamesallen I'm not sure that I agree with some of your points on the config menu, but I think it is outside the scope of this PR, so we can discuss it when the config menu system gets written, or in discord.